### PR TITLE
Fix output labels for AudioFlamingo3 (and related) models

### PR DESCRIPTION
**Pull request by ebezzam**
[CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=47112)

# What does this PR do?

Fix bug in prepared output labels, as mentioned [here](https://github.com/huggingface/transformers/pull/47111/changes#r3529962850), after [this refactor](https://github.com/huggingface/transformers/pull/45493) I think?

Script comparing before/after fix
```python
from transformers import AudioFlamingo3ForConditionalGeneration, AutoProcessor


model_id = "nvidia/audio-flamingo-3-hf"
processor = AutoProcessor.from_pretrained(model_id)
model = AudioFlamingo3ForConditionalGeneration.from_pretrained(model_id, device_map="auto")
model.train()

conversation = [
    [
        {
            "role": "user",
            "content": [
                {"type": "text", "text": "Transcribe the input speech."},
                {
                    "type": "audio",
                    "path": "https://huggingface.co/datasets/nvidia/AudioSkills/resolve/main/assets/WhDJDIviAOg_120_10.mp3",
                },
            ],
        },
        {
            "role": "assistant",
            "content": [
                {
                    "type": "text",
                    "text": "The transcription of the audio is 'summer follows spring the days grow longer and the nights are warm'.",
                }
            ],
        },
    ],
    [
        {
            "role": "user",
            "content": [
                {
                    "type": "text",
                    "text": "This track feels really peaceful and introspective. What elements make it feel so calming and meditative?",
                },
                {
                    "type": "audio",
                    "path": "https://huggingface.co/datasets/nvidia/AudioSkills/resolve/main/assets/FPSbCAANfbJLVSwD.mp3",
                },
            ],
        },
        {
            "role": "assistant",
            "content": [
                {"type": "text", "text": "The transcription of the audio is 'some transcription of the audio'."}
            ],
        },
    ],
]

inputs = processor.apply_chat_template(
    conversation,
    tokenize=True,
    add_generation_prompt=True,
    return_dict=True,
    processor_kwargs={"output_labels": True},
).to(model.device, dtype=model.dtype)

# What the model is actually trained to predict: decode the unmasked label positions
for i, labels in enumerate(inputs["labels"]):
    print(f"Labels {i}:", repr(processor.tokenizer.decode(labels[labels != -100])))

loss = model(**inputs).loss
loss.backward()
print("Loss:", loss.item())


"""
BEFORE

Labels 0: '!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!'
Labels 1: '!!!!!!!!!!!!!!$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!'
Loss: 14.966017723083496

AFTER

Labels 0: "<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n<|im_start|>user\nTranscribe the input speech.<|im_end|>\n<|im_start|>assistant\nThe transcription of the audio is 'summer follows spring the days grow longer and the nights are warm'.<|im_end|>\n<|im_start|>assistant\n"
Labels 1: "<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n<|im_start|>user\nThis track feels really peaceful and introspective. What elements make it feel so calming and meditative?<|im_end|>\n<|im_start|>assistant\nThe transcription of the audio is 'some transcription of the audio'.<|im_end|>\n<|im_start|>assistant\n"
Loss: 5.149610996246338

"""

```

### CODE REVIEW

File: src/transformers/models/audioflamingo3/processing_audioflamingo3.py [hunk 1]
```diff
@@ -99,7 +99,9 @@ def __call__(
         model_inputs = super().__call__(audio=audio, text=text, **kwargs)
 
         if output_labels:
-            labels = model_inputs.pop("mm_token_type_ids")
```
Comment (ebezzam) [inline src/transformers/models/audioflamingo3/processing_audioflamingo3.py:102]: the fix. the other two are generated via modular

File: src/transformers/models/glmasr/processing_glmasr.py [hunk 2]
```diff
@@ -105,7 +105,9 @@ def __call__(
         model_inputs = super().__call__(audio=audio, text=text, **kwargs)
 
         if output_labels:
-            labels = model_inputs.pop("mm_token_type_ids")
+            mm_token_type_ids = model_inputs.pop("mm_token_type_ids")
+            labels = model_inputs["input_ids"].clone()
```
Comment (zucchini-nlp) [inline src/transformers/models/glmasr/processing_glmasr.py:109]: omg, right! Thanks for catching that

**Review by zucchini-nlp — APPROVED** [review]

**Comment by zucchini-nlp** [comment]
should we add a test in processor tester?  would be nice so we can catch bugs early on

File: tests/models/audioflamingo3/test_processing_audioflamingo3.py
Comment (zucchini-nlp) [inline tests/models/audioflamingo3/test_processing_audioflamingo3.py:1]: nice!

File: src/transformers/models/musicflamingo/modular_musicflamingo.py [hunk 3]
```diff
@@ -145,7 +145,7 @@ def replace_audio_token(self, audio_inputs: dict, audio_idx: int) -> str:
         return self.audio_bos_token + self.audio_token * num_audio_tokens + self.audio_eos_token
 
     @property
-    def audio_ids(self):
+    def audio_token_ids(self):
```
Comment (ebezzam) [inline src/transformers/models/musicflamingo/modular_musicflamingo.py:158]: audio_token_ids is expected [here](https://github.com/huggingface/transformers/blob/eee480d59810135f45280f8db99f14d0136bed82/src/transformers/processing_utils.py#L914)

File: src/transformers/models/audioflamingo3/processing_audioflamingo3.py (hunk 1 above)
Comment (vasqu) [reply src/transformers/models/audioflamingo3/processing_audioflamingo3.py:102]: Not necessarily here but I expect quite a few to do the same ish re shifting based on the input --> maybe a util?

File: src/transformers/models/musicflamingo/modular_musicflamingo.py (hunk 3 above)
Comment (vasqu) [reply src/transformers/models/musicflamingo/modular_musicflamingo.py:158]: should we still be a bit careful and make some BC deprecation

File: tests/models/musicflamingo/test_processing_musicflamingo.py
Comment (vasqu) [inline tests/models/musicflamingo/test_processing_musicflamingo.py:1]: Nit if we are already testing audio flamingo and music flamingo we should also glm asr imo

**Review by vasqu — APPROVED** [review]
Just nits but overall looks good, thanks 🙏

File: tests/models/musicflamingo/test_processing_musicflamingo.py
Comment (ebezzam) [reply tests/models/musicflamingo/test_processing_musicflamingo.py:1]: GLM ASR doesn't have processor tests 🥲 but adding some!
